apps/webapp/src/components/staking/helpers.ts

```diff
@@ -0,0 +1,28 @@
+import {
+  BondingState_BondingStateEnum,
+  ValidatorInfo,
+  ValidatorState_ValidatorStateEnum,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/component/stake/v1/stake_pb';
+import {
+  getBondingStateEnumFromValidatorInfo,
+  getFundingStreamsFromValidatorInfo,
+  getRateBpsFromFundingStream,
+  getStateEnumFromValidatorInfo,
+} from '@penumbra-zone/types';
+
+export const getStateLabel = (validatorInfo: ValidatorInfo): string =>
+  ValidatorState_ValidatorStateEnum[getStateEnumFromValidatorInfo(validatorInfo)];
+
+export const getBondingStateLabel = (validatorInfo: ValidatorInfo): string =>
+  BondingState_BondingStateEnum[getBondingStateEnumFromValidatorInfo(validatorInfo)];
+
+const toSum = (prev: number, curr: number) => prev + curr;
+
+/**
+ * Given a `ValidatorInfo`, returns the sum of all commission rates in terms of
+ * [basis points](https://en.wikipedia.org/wiki/Basis_point).
+ */
+export const calculateCommission = (validatorInfo: ValidatorInfo): number => {
+  const fundingStreams = getFundingStreamsFromValidatorInfo(validatorInfo);
+  return fundingStreams.map(getRateBpsFromFundingStream).reduce(toSum);
```

Comment:
This is the total commission, per the logic @erwanor described [here](https://github.com/penumbra-zone/web/issues/540#issuecomment-1953271544) and per pcli's implementation [here](https://github.com/penumbra-zone/penumbra/blob/48e32b4/crates/bin/pcli/src/command/query/validator.rs#L109-L115).